src/bun.js/test/jest.zig

```diff
@@ -481,6 +803,49 @@ pub const Expect = struct {
         pub const Set = std.EnumSet(Op);
     };
 
+    pub fn getSnapshotName(this: *Expect, allocator: std.mem.Allocator, hint: string) ![]const u8 {
+        const test_name = this.scope.tests.items[this.test_id].label;
+
+        var length: usize = 0;
+        var curr_scope: ?*DescribeScope = this.scope;
+        while (curr_scope) |scope| {
+            if (scope.label.len > 0) {
+                length += scope.label.len + 1;
+            }
+            curr_scope = scope.parent;
+        }
+        length += test_name.len;
+        if (hint.len > 0) {
+            length += hint.len + 2;
+        }
+
+        var buf = try allocator.alloc(u8, length);
```

Comment:
what happens if unicode in the test name?

Reply:
removed `withEncoding()` to handle unicode and added a test